### DISCUSSION TITLE
Linking doors to buttons and assigning wait times to door waypoints.

### DISCUSSION THREAD

**Discussion by trevormoney1** [FakeFishGames/Barotrauma#8148] [Ideas and suggestions]
Right now bots are absolutely stupid when it comes to doors and effectively monkey typewrite about them. This is a suggestion that should be in theory fairly simple to implement and will make AI not only good at using custom subs, but likely a mainstay of them.

Allow doors to be linked specifically to buttons. When linked, AI trying to pass through a door, who do not find a direct wire from a button to said door, will then try to find a linked button, and will path to any that they can reach. They will then press the button.

Another thing that goes with this is the ability to edit a property on door waypoints called "Wait time" which will make AI wait at the door after pressing a corresponding button for the specified amount of time. This is so that they don't get discouraged waiting for, say, an airlock to drain.

Please take this seriously because it would make so many more submarines reasonably playable with smaller populations.

**Comment by ghost** [comment]
I thought you could already link buttons and doors?

**Comment by MaaiG** [comment]
Yep!
Related [topic](https://github.com/Regalis11/Barotrauma/discussions/7424).

>Another thing that goes with this is the ability to edit a property on door waypoints called "Wait time" which will make AI wait at the door after pressing a corresponding button for the specified amount of time. This is so that they don't get discouraged waiting for, say, an airlock to drain.

Interesting solution! Maybe perfect solution for last [paragraph](https://github.com/Regalis11/Barotrauma/discussions/7424#discussioncomment-1728373).

>the ability to edit a property on door waypoints called "Wait time"

Perhaps this should be set at the door...?

**Comment by Oukido** [comment]
Also related issue which should cover the linking: https://github.com/Regalis11/Barotrauma/issues/8093

And yes please make that property for the doors
